test-coverage.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
```

Comment:
I'm not too big a fan of this file, I haven't read it, but it looks like a lot for something that is just meant to run tests.

Reply:
What do you suggest we do? This helps me when I'm working on any python project, so I don't have to remember all the commands and config files to call. We could move all of this into the make file, but that would bulk it up.

Happy to hear suggestions.

Reply:
Since we will always write tests when we implement something or fix something, having something before us that gives us all the necessary info with just one command (and we don't have to remember all the `pytest` runner args) - thought this could be something to help the developer.

Reply:
I think you should use tox then :)
It will trigger everything for you
https://tox.readthedocs.io/en/latest/